File: 算法竞赛进阶指南/0x00基本算法/AcWing89.cpp
#include <iostream>
using namespace std;

int power(int a, int b, int p) {
  int ans = 1 % p;
  for (; b; b>>=1) {
    if (b&1) ans = (long long)ans * a % p;
    a = (long long)a * a % p;
  }
  return ans;
}

int main() {
  int a, b, p;
  cin >> a >> b >> p;
  cout << power(a, b, p);
  return 0;
}

File: 算法竞赛进阶指南/0x00基本算法/AcWing89.py
def power(a, b, p):
    ans = 1 % p
    while b:
        if b&1: ans = ans * a % p
        a = a * a % p
        b >>= 1
    return ans

def main():
    s = input()
    a, b, p = [int(x) for x in s.split()]
    print(power(a, b, p))
    
main()